servicetalk-opentelemetry/src/main/java/io/servicetalk/opentelemetry/ScopeTracker.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicetalk.opentelemetry;
+
+import io.servicetalk.concurrent.api.Single;
+import io.servicetalk.concurrent.api.TerminalSignalConsumer;
+import io.servicetalk.http.api.HttpResponseMetaData;
+import io.servicetalk.http.api.StreamingHttpResponse;
+import io.servicetalk.http.utils.BeforeFinallyHttpOperator;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Scope;
+
+import javax.annotation.Nullable;
+
+import static io.servicetalk.http.api.HttpResponseStatus.StatusClass.SERVER_ERROR_5XX;
+import static java.util.Objects.requireNonNull;
+
+class ScopeTracker implements TerminalSignalConsumer {
```

Comment:
Consider making it `final`

Reply:
This class cannot be final. I override one of its methods in the OpenTelemetryHttpServerFilter with an anonnymous instance.